### ISSUE TITLE
bug(oxlint): `--ignore-path` not respected when a config is provided

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5021]
Tested on [teambit/bit](https://github.com/teambit/bit)'s repository.

repro steps:
```sh
# run w/o config. These two should do the same thing
oxlint
oxlint --ignore-path .eslintignore
```

Then, create a simple `.oxlintrc.json`:
```json
{
  "env": {
    "browser": true,
    "node": true,
    "mocha": true
  },
  "rules": {
    "no-console": "error",
    "no-unused-vars": ["error", { "args": "after-used" }],
    "no-shadow": "off"
  }
}
```

And run `oxlint` again with the config:
```sh
oxlint --config .oxlintrc.json
oxlint --config .oxlintrc.json --ignore-path .eslintignore
```

## Expected Behavior
Files in `e2e/fixtures` (and all other ignored directories) do not get linted
## Actual Behavior
`.eslintignore` is registered with `WalkBuilder` but never actually respected.

**Comment by shulaoda** [comment]
Everything seems to be working properly, it seems like this issue has been resolved?

**Comment by spacepumpkin** [comment]
Still seeing this bug on v0.9.10. `--ignore-path` doesn't work when a config file is provided.

**Comment by jordan-boyer** [comment]
I wanted to tackle this issue but it seems to me that it's working fine with v0.9.10

I think we need more info to try reproduce. 

Here is my steps to reproduce on the teambit/bit repo

- Run oxlint -> found 197 warnings and 3 errors 
- Run oxlint --ignore-path .eslintignore -> found 197 warnings and 3 errors 
- Add DonIsaac config on repo and Run oxlint ->  Found 50 warnings and 182 errors. (there is 7 errors link to src/e2e-helper folder) 
- Add src/e2e-helper to .eslintignore and Run oxlint -> found 50 warnings and 175 errors. 

you can see here that  7 errors is deleted on the last try because we did add and ignore in .eslintignore

**Comment by Boshen** [comment]
> Still seeing this bug on v0.9.10. `--ignore-path` doesn't work when a config file is provided.

A reproduction is appreciated.
